src/index.js

```diff
@@ -1,39 +1,49 @@
 import {queries, waitForElement} from 'dom-testing-library'
 
-const commands = Object.keys(queries)
-  .filter(
-    queryName =>
-      queryName.startsWith('getBy') || queryName.startsWith('getAllBy'),
-  )
-  .map(queryName => {
-    return {
-      name: queryName,
-      command: (cy, ...args) => {
-        const queryImpl = queries[queryName]
-        const commandImpl = doc =>
-          waitForElement(() => queryImpl(doc, ...args), {container: doc})
-        const thenHandler = new Function(
-          'commandImpl',
-          `
+const commands = Object.keys(queries).map(queryName => {
+  return {
+    name: queryName,
+    command: (cy, ...args) => {
+      const queryImpl = queries[queryName]
+      const baseCommandImpl = doc =>
+        waitForElement(() => queryImpl(doc, ...args), {
+          container: doc,
+          timeout: 3000,
```

Comment:
It's not related directly to the subject of this PR, but as you're adding hardcoded `timeout: 3000` I would suggest adding a way to pass these defaults to the command, e.g. use the argument that follows query arguments, something like `args[queryImpl.length]` (hacky generic way, `queryImpl` arity plus one) as an `options` object of structure like `{ timeout: 3000, waitForElementOptions: { ... } }` (`timeout` as a top-level option because it may be used in multiple lower-level calls, not just in `waitForElement` call).

P.S. Cypress default timeout is 4000, not 3000.

Reply:
If we do the same timeout as Cypress, we run the risk of Cypress timing out before us. Could we make the Cypress timeout 5000ms for this command to deal with that issue?